cmake/modules/FindSpdlog.cmake

```diff
@@ -35,6 +35,12 @@ if(ENABLE_INTERNAL_SPDLOG)
 
   set(SPDLOG_LIBRARY ${CMAKE_BINARY_DIR}/${CORE_BUILD_DIR}/lib/libspdlog.a)
   set(SPDLOG_INCLUDE_DIR ${CMAKE_BINARY_DIR}/${CORE_BUILD_DIR}/include)
+
+  # make sure we use the internally built fmt
+  if(ENABLE_INTERNAL_FMT)
+    set(FMT_ROOT ${CMAKE_BINARY_DIR}/${CORE_BUILD_DIR})
```

Comment:
Why only set this if internal fmt is enabled? Always setting it doesn't cause any harm.

Reply:
I'm not sure of the side effects. But any library found in the internal build directory should probably have precedence over system libraries.

Reply:
I removed the `if(ENABLE_INTERNAL_FMT)`